_shared/project/.github/workflows/ci.yml

```diff
@@ -14,14 +14,21 @@ jobs:
     needs: tests
     {% endif %}
     runs-on: ubuntu-latest
-    {% if cookiecutter.get("db") == "yes" %}
+    {% if job in ["tests", "functests"] and has_services() %}
     services:
+      {% if cookiecutter.get("db") == "yes" %}
       postgres:
-        image: postgres:11.5-alpine
+        image: postgres:{{ cookiecutter.get("__postgres_version", "11.5-alpine") }}
```

Comment:
Yep.

And the thing is you can't have optional questions. So if things like `__postgres_version` and `__postgres_port` didn't have the `__`'s then even if the user said "no" to Postgres (which is the default) it would still go on to ask you what Postgres version and port you wanted :( Hence the approach I've taken here is that `__postgres_version` and `__postgres_port` are private options (so the interactive UI never asks you for them) but they have default values (including that value being randomly generated in the case of the port number). A similar approach has been used for other dependent options